src/seedu/addressbook/Main.java

```diff
@@ -105,12 +110,15 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
+            throw new StorageOperationException("Storage file is read-only. Please allow write access.");
```

Comment:
Good job defining your own exception message (: 

You do not have to throw another exception to `runCommandLoopUntilExitCommand()` because you can handle this exception in `executeCommand()` by doing `ui.showToUser("Storage file is read-only. Please allow write access.")` and return `new CommandResult("")`. Showing the user the error message once is sufficient.